src/gemdat/volume.py

```diff
@@ -122,6 +129,36 @@ def find_peaks(
 
         return coords[:, 0:3].astype(int)
 
+    def nearest_structure_reference(self, structure: Structure):
+        """Find distance and index of the nearest site of the structure for
+        each voxel using a KD-tree.
+
+        Parameters
+        ----------
+        structure : pymatgen.core.structure.Structure
+            Structure of the material to use as reference for nearest site
+        """
+        # In order to accomodate the periodicity, include the images of the structure sites
+        periodic_structure = []
+        periodic_ids: list[int] = []
+        images = np.mgrid[-1:2, -1:2, -1:2].reshape(3, -1).T
+        for dx, dy, dz in images:
+            periodic_structure.extend(
+                self.lattice.get_fractional_coords(
+                    structure.cart_coords + np.array([
+                        self.lattice.a * dx, self.lattice.b *
+                        dy, self.lattice.c * dz
+                    ]) * self.resolution))
```

Comment:
I can't understand what this is trying to do. It's going from fractional to cartesian and back to fractional... Is this just adding the translations? 

Why not something like this:

```suggestion
            periodic_structure.extend(structure.frac_coords + np.array([dx, dy, dz])
```